Kudu.Core/Deployment/DeploymentManager.cs

```diff
@@ -327,6 +344,32 @@ public static ChangeSet CreateTemporaryChangeSet(string authorName = null, strin
             };
         }
 
+        private async Task RestartMainSiteAsync(DeploymentInfoBase deploymentInfo)
```

Comment:
I believe this is fire and forget but if this request returns a non success code, should this fail the deployment ?

Reply:
Yeah, I was thinking of the same. I pushed a change to return a failure. There's no good way to ensure that the operation is atomic though.

Reply:
Yeah, I was thinking of the same. I pushed a change to return a failure. There's no good way to ensure that the operation is atomic though.

---
In reply to: [332693077](https://github.com/projectkudu/kudu/pull/3050#discussion_r332693077) [](ancestors = 332693077)